### DISCUSSION TITLE
Re-implementing Spinning Trains for Classic Junior Coaster

### DISCUSSION THREAD

**Discussion by Derpercoaster** [OpenRCT2/OpenRCT2#18282] [Feature requests]
In RCT1, the steel mini coaster had trains with cars which spun on curves, but could not traverse banked track. These trains were removed in RCT2, but they have not been implemeted in OpenRCT2. How easy would it be to re-implement them?

**Comment by Generalcamo** [comment]
The complication that was brought up years ago (see: https://github.com/OpenRCT2/objects/issues/138) was that the rotation code assumes you have 32 sprites for orientation, while the spinning trains only have 16. I'm not sure if this has been resolved yet.

**Reply by Derpercoaster** [reply]
I think there's only 16 because the spinning train cars don't have a defined "front" side, which means the game can just use the same set of sprites twice to justify 32 different sprite orientations.

**Reply by Generalcamo** [reply]
The original RCT only had 16 sprites for orientation for any car. RCT2 doubled this for smoother animation. I believe this was a complication with porting the original RCT1 trains as well before that was eventually resolved.

**Reply by spacek531** [reply]
It was resolved for regular trains but spinning trains have special paint code which has not been adapted.

**Reply by TheClaudeQC** [reply]
Spinning trains have a interesting way, each time it dose a 1/4 of a rotation, it replace back to the 1st sprite used, but pivoting the guests from their seats
